tests/scripts/generatedirsproj.csx

```diff
@@ -0,0 +1,208 @@
+using System.IO;
+using Microsoft.Build.Construction;
+using Microsoft.Build.Evaluation;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Tasks;
+using Microsoft.Build.Utilities;
+
+var start = DateTime.UtcNow;
+var projectTemplateName = "dirs.proj.xml";
+var outputProjectPath = "..\\src\\dirs.proj";
+
+Console.WriteLine();
+Console.WriteLine($" Starting generation of {Path.GetFullPath(outputProjectPath)} project");
+Console.WriteLine();
+
+// Prepare and run CreateItem task to glob project files
+var projTempl = ProjectRootElement.Open(projectTemplateName);
+CreateItem task = new CreateItem();
+
+task.Include = new ITaskItem[]
+{
+    new TaskItem(@"..\src\**\*.csproj"),
+    new TaskItem(@"..\src\**\*.ilproj")
+};
+
+ProjectTargetElement disabledTarget = null; 
+foreach(var target in projTempl.Targets)
+{
+    if (target.Name == "ResolveDisabledProjects")
+    {
+        disabledTarget = target;
+        break;
+    }
+}
+
+Dictionary<string, string> conditionalExcludes = new Dictionary<string, string>();
+if (disabledTarget != null)
+{
+    foreach(var itemsGroup in disabledTarget.ItemGroups)
+    {
+        if (itemsGroup.Label == "DisabledProjectsItemGroup")
+        {
+            List<ProjectItemElement> items = new List<ProjectItemElement>(itemsGroup.Items);
+            for(int i = 0; i < items.Count; )
+            {
+                if (!String.IsNullOrEmpty(items[i].Condition))
+                {
+                    conditionalExcludes.Add(items[i].Include, items[i].Condition);
+                    items.RemoveAt(i);
+                    continue;
+                }
+                i++;
+            }
+        }
+    }
+}
+
+task.Exclude = new ITaskItem[]
+{
+      new TaskItem(@"..\src\TestWrappers*\**\*.csproj"),
```

Comment:
I don't like duplicating these sets of excluded test projects. We should figure out a way to share.

Reply:
I am working on more comprehensive fix which will address @AndyAyersMS feedback and will include fix to the above duplication of excluded test projects